File: frontend/src/types/advert.ts
import type { AdvertStatus, City, MusicBranch } from './enums';

// Sahnem.Business/DTOs/Advert/AdvertResponseDto.cs
export interface Advert {
  id: number;
  creatorId: number;
  creatorName: string;
  creatorRole: 'Organizer' | 'Venue';
  title: string;
  description: string;
  city: City;
  district?: string;
  address: string;
  equipmentProvided: boolean;
  equipmentNote?: string;
  eventTime: string;
  budget: number;
  minimumExperienceYears?: number;
  branch?: MusicBranch;
  status: AdvertStatus;
  applicationDeadline: string;
  createdDate: string;
  offerCount: number;
}

// Sahnem.Business/DTOs/Advert/AdvertCreateDto.cs (+ AdvertUpdateDto — alanlar birebir aynı)
export interface AdvertCreateInput {
  title: string;
  description: string;
  city: City;
  district?: string;
  address: string;
  equipmentProvided: boolean;
  equipmentNote?: string;
  eventTime: string;
  budget: number;
  minimumExperienceYears?: number;
  branch?: MusicBranch;
  applicationDeadline: string;
}

export type AdvertUpdateInput = AdvertCreateInput;

File: backend/Sahnem.Core/Entities/Advert.cs
using Sahnem.Business.Enums;
using Sahnem.Core.Enums;

namespace Sahnem.Core.Entities
{
    
    public class Advert : BaseEntity
    {
        public int CreatorId {get; set;}
        public string Title {get; set;}
        public string Description {get; set;}
        public City City{get; set;}
        public string? District{get; set;}
        public string Address {get; set;}
        public bool EquipmentProvided { get; set; }
        public DateTime EventTime {get; set;}
        public decimal Budget {get; set;}
        public int? MinimumExperienceYears{get; set;}
        public AdvertStatus Status {get; set;}
        public DateTime ApplicationDeadline { get; set; }

        public virtual AppUser Creator {get; set;}
        public virtual ICollection<Offer> Offers {get; set;} = new List<Offer>();

    }
}